README.md

```diff
@@ -13,96 +13,104 @@
-   &state=12345
-   &redirect_uri=https://login.microsoftonline.com
-   ```
+9. Use the **Select permissions** search box to search for the following permissions: Directory.Read.All, Group.ReadWrite.All, Mail.Read, Mail.ReadWrite, and User.Read.All. Select the check box for each permissions as it appears (note that the permissions will not remain visible in the list as you select each one). Select the **Add permissions** button at the bottom of the panel.
 
-After signing in, click **Accept** in the consent page.  You can then close the browser.  Now that you've pre-consented, you can try running the console sample in app mode. 
+10. Choose the **Grant admin consent for [tenant name]** button. Select **Yes** for the confirmation that appears.
 
-## Build and run the sample 
+## Build and run the sample
 
 1. Open the sample solution in Visual Studio.
 2. Press F5 to build and run the sample. This will restore the NuGet package dependencies and open the console application.
 3. Select User mode to run the application with **delegated permissions** only. Select App mode to run the application with **application permissions** only. Select both to run using both types of permissions.
 4. When you run User mode, you'll be prompted to sign in with an account on your Office 365 tenant and consent to the permissions that the application requests. If you want to run the groups-related operations in the **UserMode** class, you'll need to uncomment the **GetDetailsForGroups** method in the UserMode.cs file and the **Group.Read.All** scope in the AuthenticationHelper.cs file. After you make those changes only an admin will be able to sign in and consent. Otherwise, you can sign in and consent with a non-admin user.
 5. When you run App mode, the application will begin performing a number of common groups-related tasks that only an admin can do. Since you've already authorized the application to make these operations, you won't be prompted to sign in and consent.
-   
+
 ## Questions and comments
 
 We'd love to get your feedback about the Microsoft Graph API Console App. You can send your questions and suggestions in the [Issues](https://github.com/microsoftgraph/console-csharp-snippets-sample/issues) section of this repository.
 
 Questions about Microsoft Graph development in general should be posted to [Stack Overflow](https://stackoverflow.com/questions/tagged/microsoftgraph). Make sure that your questions or comments are tagged with [microsoftgraph].
 
-## Contributing ##
+## Contributing
 
 If you'd like to contribute to this sample, see [CONTRIBUTING.MD](/CONTRIBUTING.md).
 
 This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.
   
 ## Additional resources
 
```

Comment:
Should update these links to the docs links. I know they redirect, but it's better to have them right

Reply:
Thanks, fixed links at both beginning and end of the doc. Also fixed a stray ARP link. Merging.